app/model/command/ImportCampaignFromCAPICommand.scala

```diff
@@ -62,6 +46,61 @@ case class ImportCampaignFromCAPICommand(
     }.getOrElse(Nil)
   }
 
+  def buildContentItems(apiContent: List[ApiContent], campaignId: String): List[ContentItem] = apiContent.map{ apic =>
+    ContentItem(
+      campaignId = campaignId,
+      id = apic.fields.flatMap(_.internalComposerCode).getOrElse(UUID.randomUUID().toString),
+      `type` = deriveContentType(apic),
+      composerId = apic.fields.flatMap(_.internalComposerCode),
+      path = Option(apic.id),
+      title = apic.webTitle,
+      isLive = apic.fields.flatMap(_.isLive).getOrElse(false),
+      atoms = buildAtomList(apic)
+    )
+  }
+
+  def updateCampaignAndContent(apiContent: List[ApiContent], hostedTag: Tag, campaign: Campaign): Option[Campaign] = {
+    val contentItems = buildContentItems(apiContent, campaign.id)
+
+    contentItems.foreach( CampaignContentRepository.putContent )
+
+    val startDate = apiContent.flatMap(_.fields.flatMap(_.firstPublicationDate)).sortBy(_.dateTime).headOption
+
+    val ctaAtoms = apiContent.flatMap(_.atoms.flatMap(_.cta)).flatten
+
+    val updatedCampaign = campaign.copy(
+      startDate = startDate.map{cdt => new DateTime(cdt.dateTime).withTimeAtStartOfDay()},
+      status = if(startDate.isDefined) "live" else campaign.status,
+      callToActions = ctaAtoms.map(ctaAtom => CallToAction(Some(ctaAtom.id))).distinct,
+      campaignLogo = deriveTagLogo(hostedTag)
+    )
+
+    CampaignRepository.putCampaign(updatedCampaign)
```

Comment:
It will do an insert or update depending on if the content record is already there.